src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.jsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import Tooltip from "../Tooltip/Tooltip";
+import Dialog from "../Dialog/Dialog";
+import AvatarGroupCounterTooltipContent from "./AvatarGroupCounterTooltipContent";
+import Avatar from "../Avatar/Avatar";
+import PropTypes from "prop-types";
+
+const AvatarGroupCounterTooltipContainer = ({ children, avatars, type, className, counterTooltipCustomProps }) => {
+  const counterTooltipContentExists = !!avatars?.length;
+  const counterTooltipContentComponent = (
+    <AvatarGroupCounterTooltipContent avatars={avatars} type={type} className={className} />
+  );
+
+  if (!counterTooltipContentExists && !counterTooltipCustomProps) {
+    return children;
+  }
+
+  if (counterTooltipCustomProps) {
```

Comment:
no need for if else, you can just write in the tooltip props like this:
{...counterTooltipCustomProps} and set default as {}
and regarding content do something like this: 
content={counterTooltipCustomProps?.content || counterTooltipContentComponent}

Reply:
I've refactored it, but in a different way, check it out
If counterTooltipCustomProps have been passed even without content inside we don't want to display counterTooltipContentComponent and don't want it to be rendered